kong/tools/batch_queue.lua

```diff
@@ -195,8 +198,7 @@ process = function(premature, self, batch)
   if #self.batch_queue > 0 then -- more to process?
     ngx_log(DEBUG, fmt("processing oldest data, %d still queued",
                    #self.batch_queue - 1))
-    local oldest_batch = remove(self.batch_queue, 1)
-    schedule_process(self, oldest_batch, next_retry_delay)
+    schedule_process(self, next_retry_delay)
```

Comment:
I believe the refactor will make the previous debug message incorrect - since now we remove before logging, the `- 1` isn't needed any more:

```
ngx_log(DEBUG, fmt("processing oldest data, %d still queued",
                   #self.batch_queue))
```